i18n/en/docusaurus-plugin-content-docs/current/reference/layers.mdx

```diff
@@ -0,0 +1,174 @@
+
+This layer, unlike others, does not consist of slices, and instead consists of segments directly.
+
+**Content examples**:
+
+* UI kit
+* API client
+* Code working with browser APIs
+* Routing primitives (e.g., authentication-protected routes)
+
+### Entities
+
+Concepts from the real world that form together the essence of the project. Commonly, these are the terms that the business uses to describe the product.
+
+Each slice in this layer contains static UI elements, data stores and CRUD operations.
+
+**Slice examples**:
+
+<table>
+<thead><tr><th> For a social network </th><th> For a Git frontend (e.g., GitHub) </th></tr></thead>
+<tbody><tr><td><ul>
+<li>User</li>
+<li>Post</li>
+<li>Group</li>
+</ul></td><td><ul>
+<li>Repository</li>
+<li>File</li>
+<li>Commit</li>
+</ul></td></tr></tbody></table>
+
+
+:::tip
+
+You may notice in the example of a Git frontend that a _repository_ contains _files_. This makes the repository a higher-level entity which has other entities nested inside. That is a common situation with entities, and sometimes it's hard to manage such higher-level entities without breaking the import rule on layers.
+
+Here are a few suggestions to overcome this issue:
+* The UI of entities should contain slots for places where the lower-level entities are to be inserted
+* The business logic related to entity interaction should be placed in features (most of the time)
+* The typings of database entities can be extracted to the Shared layer below, next to the API client
+
+:::
+
+### Features
+
+Actions that a user can make in the application to interact with the business entities to achieve a valuable outcome. This also includes actions that the app makes on behalf of the user to produce value for them.
+
+Each slice in this layer can contain interactive UI elements, internal state and API calls that enable value-producing actions.
+
+**Slice examples**:
+
+<table>
+<thead><tr><th> For a social network </th><th> For a Git frontend (e.g., GitHub) </th><th> Actions on behalf of users </th></tr></thead>
+<tbody><tr><td><ul>
+<li>Authenticate</li>
+<li>Create a post</li>
+<li>Join a group</li>
+</ul></td><td><ul>
+<li>Edit a file</li>
+<li>Leave a comment</li>
+<li>Merge branches</li>
+</ul></td><td><ul>
+<li>Detect dark mode</li>
+<li>Perform background computation</li>
+<li>User-Agent-based actions</li>
+</ul></td></tr></tbody></table>
+
+### Widgets
+
+Self-sufficient UI blocks emerged from the composition of lower-level units like entities and features.
+
+This layer provides a way to fill in the slots left in the UI of Entities with other Entities and interactive elements from Features. Therefore, it is common not to have business logic on this layer, instead keeping it in Features. Each slice in this layer contains ready-to-use UI components and sometimes non-business logic such as gestures, keyboard interaction, etc.
```

Comment:
This kind of definition doesn't cover a use case of `rich data-tables`. They has a lot of business logic and its not clear where to put them.

Can we add to the example some `rich data-table`, and some notes what it's `common to have` business logic here?

Reply:
What do you mean by a rich data table? Do you just mean search and sorting or do you mean inline forms?

I'm not sure I understand your use case about having _business_ logic on this layer, but I totally see how one could have non-business logic on this layer.

Reply:
- custom view/edit columns(integration with form) 
- custom action columns (features integration)
- custom rich filters/search
- regular features integration
- rich view model (sub rows / Modal-view)